packages/website/src/snippets.ts

```diff
@@ -253,36 +256,127 @@ export const umdScriptTags: ScriptTagsType = (
+      #nlx-multimodal-resume { 
+        animation: fadeInKeyframes 0.6s normal forwards ease-in-out;
+        position: fixed;
+        bottom: 0; 
+        left: 0;
+        right: 0;
+        z-index: 10000;
       }
-    );
-  });
-</script>
+
+      #nlx-multimodal-resume button {
+        display: block;
+        width: 100%;
+        padding: 15px 10px;
+        font-size: 12px;
+        letter-spacing: 0.5px;
+        font-weight: 600;
+        text-transform: uppercase;
+        text-align: center;
+        border: 0;
+        background-color: rgba(38, 99, 218, 1);
+        color: white;
+        cursor: pointer;
+      }
+        
+      #nlx-multimodal-resume button:hover {
+        background-color: rgba(30, 86, 196, 1); 
+      }
+    </style>`
+      : ""
+  }
+    ${umdScriptTags.journeyManager}
+    <script>
+      // Get conversation ID from the URL, assuming it is included as a 'cid' search param (and save in session storage)
+      // On subsequent page loads, retrieve from session storage assuming it was saved in the past half an hour
+      const getCid = () => {
+        const localStorageKey = "nlx-multimodal-session";
+        const saveToSessionStorage = (conversationId) => {
+          sessionStorage.setItem(localStorageKey, JSON.stringify({
+            conversationId,
+            timestamp: new Date().getTime()
+          }));
+        };
+        const retrieveFromSessionStorage = () => {
+          try {
+            const val = sessionStorage.getItem(localStorageKey);
+            if (!val) {
+              return null;
+            }
+            const parsed = JSON.parse(val);
+            if (parsed != null && typeof parsed == "object" && new Date().getTime() - parsed.timestamp < 30 * 60 * 1000) {
+              return parsed.conversationId;
+            }
+          } catch(_err) {
+            return null;
+          }
+        }
+        const params = new URLSearchParams(window.location.search);
+        const cidFromParams = params.get("cid");
+        if (cidFromParams) {
+          saveToSessionStorage(cidFromParams);
+          return cidFromParams;
+        }
+        return retrieveFromSessionStorage();
+      };
+
+      const triggers = {};
+
+      window.addEventListener("DOMContentLoaded", () => {
+        nlxai.journeyManager.run({
+          config: {
+            // hard-coded params
+            apiKey: "REPLACE_WITH_API_KEY",
+            workspaceId: "REPLACE_WITH_WORKSPACE_ID",
+            journeyId: "REPLACE_WITH_JOURNEY_ID",
+            // dynamic params
+            conversationId: getCid(),
+            languageCode: "REPLACE_WITH_LANGUAGE_CODE",
+          },
+          // Triggers object obtained from the journey step configuration
+          triggers: {},
+          // Send a custom step if a digression is detected
+          onDigression: (client) => {
+${
+  digressionButton
+    ? `            const container = document.createElement("div");
```

Comment:
is there some explanation in the docs what the digression button does?

Reply:
No this is custom functionality, we're just code-generating some behavior for the user to copy-paste onto their page, stuff that is not included in the package.

Digression detection, on the other hand, is a real thing that we will have to document proper, but I have a feeling it will change around quite a bit.